lib/cli-engine.js

```diff
@@ -391,12 +392,14 @@ function getCacheFile(cacheFile, cwd) {
  */
 function CLIEngine(options) {
 
+    options = (options instanceof Object) ? assign(options) : {};
+    options.cwd = options.cwd || process.cwd();
+
     /**
      * Stored options for this instance
      * @type {Object}
      */
-    this.options = assign(Object.create(defaultOptions), options || {});
-
+    this.options = assign(defaultOptions, options);
```

Comment:
try running some cli tests and observe the values getting set. It will happen when the call come from cli into cli-engine. I have to try myself to make sure I am 100% correct.